web/frontend/src/pages/election/components/DKGStatusRow.tsx

```diff
@@ -50,13 +62,107 @@ const DKGStatusRow: FC<DKGStatusRowProps> = ({
     signal: signal,
   };
 
+  const [showModal, setShowModal] = useState(false);
+
+  // Notify the parent each time our status changes
+  useEffect(() => {
+    notifyDKGState(node, InternalDKGInfo.fromStatus(status));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [status]);
```

Comment:
If a node fails to initialize once, it get the status Failed. If it fails another time to initialize, then it gets stuck in Initializing, because the status does not change, so the parent are not notified. This normally works:

```suggestion
  }, [DKGLoading]);
```

Reply:
True. Then I should manually handle this case and call `notifyDKGState` once I get the second - same - initialization failure.

Reply:
Yep !

Reply:
Even if the state doesn't change, the node notifies the parent with the `notifyLoading` callback. So the parent can still make the check.